src/pages/non-pass-finalisation/cat-be/non-pass-finalisation.cat-be.page.ts

```diff
@@ -0,0 +1,190 @@
+import { Store, select } from '@ngrx/store';
+import { StoreModel } from '../../../shared/models/store.model';
+import { CAT_B } from '../../page-names.constants';
+import { Observable } from 'rxjs/Observable';
+import { getTests } from '../../../modules/tests/tests.reducer';
+import {
+  getCurrentTest,
+  getJournalData,
+  getActivityCode,
+  getTestOutcome,
+  isTestOutcomeSet,
+  getTestOutcomeText,
+} from '../../../modules/tests/tests.selector';
+import { getCandidate } from '../../../modules/tests/journal-data/candidate/candidate.reducer';
+import {
+  getUntitledCandidateName,
+  getCandidateDriverNumber,
+  formatDriverNumber,
+} from '../../../modules/tests/journal-data/candidate/candidate.selector';
+import { NonPassFinalisationViewDidEnter } from '../non-pass-finalisation.actions';
+import { map, withLatestFrom } from 'rxjs/operators';
+import { getTestSummary } from '../../../modules/tests/test-summary/test-summary.reducer';
+import { isDebriefWitnessed, getD255 } from '../../../modules/tests/test-summary/test-summary.selector';
+import {
+  getTestSlotAttributes,
+} from '../../../modules/tests/journal-data/test-slot-attributes/test-slot-attributes.reducer';
+import { isWelshTest } from '../../../modules/tests/journal-data/test-slot-attributes/test-slot-attributes.selector';
+import {
+  ActivityCodeModel,
+  activityCodeModelList,
+} from '../../office/components/activity-code/activity-code.constants';
+import { FormGroup } from '@angular/forms';
+import { PersistTests } from '../../../modules/tests/tests.actions';
+import { OutcomeBehaviourMapProvider } from '../../../providers/outcome-behaviour-map/outcome-behaviour-map';
+import { behaviourMap } from '../../office/office-behaviour-map';
+import {
+  DebriefWitnessed,
+  DebriefUnwitnessed,
+  D255Yes,
+  D255No,
+} from '../../../modules/tests/test-summary/test-summary.actions';
+import {
+  CandidateChoseToProceedWithTestInWelsh,
+  CandidateChoseToProceedWithTestInEnglish,
+} from '../../../modules/tests/communication-preferences/communication-preferences.actions';
+import { SetTestStatusWriteUp } from '../../../modules/tests/test-status/test-status.actions';
+import { SetActivityCode } from '../../../modules/tests/activity-code/activity-code.actions';
+
+interface NonPassFinalisationPageState {
+  candidateName$: Observable<string>;
+  candidateDriverNumber$: Observable<string>;
+  isTestOutcomeSet$: Observable<boolean>;
+  testOutcome$: Observable<string>;
+  testOutcomeText$: Observable<string>;
+  activityCode$: Observable<ActivityCodeModel>;
+  displayDebriefWitnessed$: Observable<boolean>;
+  debriefWitnessed$: Observable<boolean>;
+  displayD255$: Observable<boolean>;
+  d255$: Observable<boolean>;
+  isWelshTest$: Observable<boolean>;
+}
+
+@IonicPage()
+@Component({
+  selector: 'non-pass-finalisation-cat-be-page',
+  templateUrl: 'non-pass-finalisation.cat-be.page.html',
+})
+export class NonPassFinalisationCatBEPage extends PracticeableBasePageComponent {
```

Comment:
On B+E Pages we want to change the page to extend the BasePageComponent as there is no Practice Mode.